app/controllers/services/lifecycle/service_key_manager.rb

```diff
@@ -0,0 +1,59 @@
+module VCAP::CloudController
+  class ServiceKeyManager
+    class ServiceInstanceNotFound < StandardError; end
+    class ServiceInstanceNotBindable < StandardError; end
+
+    def initialize(services_event_repository, access_validator, logger)
+      @services_event_repository = services_event_repository
+      @access_validator = access_validator
+      @logger = logger
+    end
+
+    def create_service_key(request_attrs)
+      service_instance = ServiceInstance.first(guid: request_attrs['service_instance_guid'])
+      raise ServiceInstanceNotFound unless service_instance
+      raise ServiceInstanceNotBindable unless service_instance.bindable?
+
+      service_key = ServiceKey.new(request_attrs)
+      @access_validator.validate_access(:create, service_key)
+      raise Sequel::ValidationFailed.new(service_key) unless service_key.valid?
+
+      lock_service_instance_by_blocking(service_instance) do
+        attributes_to_update = service_key.client.bind(service_key)
+        begin
+          service_key.set_all(attributes_to_update)
+          service_key.save
+        rescue
+          safe_unbind_instance(service_key)
+          raise
+        end
+      end
+
+      service_key
+    end
+
+    private
+
+    def safe_unbind_instance(service_key)
+      service_key.client.unbind(service_key)
+    rescue => e
+      @logger.error "Unable to unbind #{service_key}: #{e}"
+    end
+
+    def lock_service_instance_by_blocking(service_instance, &block)
+      return block.call unless service_instance.managed_instance?
+
+      original_attributes = service_instance.last_operation.try(:to_hash)
+      begin
+        service_instance.lock_by_failing_other_operations('update') do
+          block.call
+        end
+      ensure
+        if original_attributes
```

Comment:
The `ServiceBindingManager` is covered by the ServiceBindingController specs. We're talking about backfilling unit tests for the object. In the meantime, you guys can follow our lead and add the tests for the manager class to the controller specs.